lib/ansible/modules/monitoring/uptimerobot.py

```diff
@@ -27,124 +27,167 @@
+            - The kind of check, that will be performed on the url.
+        choices: [ "http", "dns" ]
+        version_added: '2.10'
     monitorid:
         description:
             - ID of the monitor to check.
-        required: true
+        required: false
     apikey:
         description:
             - Uptime Robot API key.
         required: true
 notes:
-    - Support for adding and removing monitors and alert contacts has not yet been implemented.
+    - Support for further functionalities, which the api provides has not yet been implemented.
+    - created and absent are added in version 2.10
 '''
 
 EXAMPLES = '''
 # Pause the monitor with an ID of 12345.
 - uptimerobot:
-    monitorid: 12345
+    name: My domain
+    url: https://www.my-domain.com
     apikey: 12345-1234512345
     state: paused
 
-# Start the monitor with an ID of 12345.
+# Create an http checking monitor for https://www.my-domain.com
 - uptimerobot:
-    monitorid: 12345
+    state: created
+    url: https://www.my-domain.com
+    name: My domain
+    check_type: http
     apikey: 12345-1234512345
-    state: started
 '''
 
 import json
-
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.six.moves.urllib.parse import urlencode
-from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.urls import Request
+from ansible.module_utils._text import to_text
 
-
-API_BASE = "https://api.uptimerobot.com/"
-
-API_ACTIONS = dict(
-    status='getMonitors?',
-    editMonitor='editMonitor?'
-)
+API_BASE = "https://api.uptimerobot.com/v2/"
 
 API_FORMAT = 'json'
 API_NOJSONCALLBACK = 1
 CHANGED_STATE = False
 SUPPORTS_CHECK_MODE = False
 
 
-def checkID(module, params):
-
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['status'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult
-
-
-def startMonitor(module, params):
-
-    params['monitorStatus'] = 1
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
-
-
-def pauseMonitor(module, params):
-
-    params['monitorStatus'] = 0
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
+class UptimeRobot:
+    def __init__(self, params, api_methods):
+        self.headers = {}
+        self.headers['Content-Type'] = "application/json"
+        self.uri = ''
+        self.api_methods = api_methods
+        self.params = params
+        self.params = self.check_dict(self.params)
+        for k, v in self.params.items():
+            if k == 'type':
+                if v == 'dns':
```

Comment:
```suggestion
                self.params[k] = 2 if v == 'dns' else 1
```

Reply:
Fixed in https://github.com/Ninjiner/ansible/commit/2165e03dd4ebc6e6b48104445526f2cfce879cf8